src/main.rs

```diff
@@ -353,71 +38,49 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 			scenario,
 			send_threshold,
 			remark,
-		} => {
-			let recipe = remark.map_or_else(
-				|| TransactionRecipe::transfer(),
-				|size| TransactionRecipe::remark(size),
-			);
-			match chain {
-				ChainType::Fake => {
-					let sink = FakeTransactionSink::new();
-					let builder = FakeTransactionBuilder::new();
-					execute_scenario(
-						sink,
-						builder,
-						scenario,
-						*unwatched,
-						*send_threshold as usize,
-						&recipe,
-					)
-					.await;
-				},
-				ChainType::Eth => {
-					let transaction_monitor =
-						if *block_monitor { Some(BlockMonitor::new(ws).await) } else { None };
+		} => match chain {
+			ChainType::Fake => {
+				unimplemented!()
+			},
+			ChainType::Eth => {
+				let mut scenario_builder = ScenarioBuilder::default()
+					.with_rpc_uri(ws.to_string())
+					.with_scenario_type(scenario.clone())
+					.with_block_monitoring()
+					.with_chain_type(chain.clone())
+					.with_send_threshold(*send_threshold as usize);
```

Comment:
We need to handle `remark` param too. 
So `.with_transfer_tx_recipe()` or `with_remark_tx_recipe` shall be called.
Also in `sub` branch.

Reply:
oh, it is handled below - but transfer needs to be default.

Reply:
Indeed. Added it for both sub & eth cases. Change here: [2319bdf](https://github.com/michalkucharczyk/tx-test-tool/pull/22/commits/2319bdf4509881f8c4c8bf186444a2b605fdb8d5)